src/Templates/Boilerplate/Bit.Boilerplate/src/Server/Boilerplate.Server.Api/Program.Services.cs

```diff
@@ -326,6 +327,34 @@ private static void AddIdentity(WebApplicationBuilder builder)
                 options.SignInScheme = IdentityConstants.ExternalScheme;
             });
         }
+        
+        if (string.IsNullOrEmpty(configuration["Authentication:MicrosoftAccount:ClientId"]) is false)
+        {
+            authenticationBuilder.AddMicrosoftAccount(options =>
+            {
+                var tenantId = configuration["Authentication:MicrosoftAccount:TenantId"]!;
+                options.ClientId  = configuration["Authentication:MicrosoftAccount:ClientId"]!;
+                options.ClientSecret  = configuration["Authentication:MicrosoftAccount:ClientSecret"]!;
+                options.SaveTokens = true;
+                
+                if (string.IsNullOrEmpty(tenantId) is false)
+                {
+                    options.AuthorizationEndpoint = $"https://login.microsoftonline.com/{tenantId}/oauth2/authorize";
+                    options.TokenEndpoint         = $"https://login.microsoftonline.com/{tenantId}/oauth2/v2.0/token";
+                }
+                options.Events = new OAuthEvents()
+                {
+                     OnRemoteFailure = context =>
+                        {
+                           
+                           var errorMessage = context.Failure?.Message ?? "An error occurred during authentication.";
```

Comment:
since we are going to show this message on a page, I suggest add them to .resx file together with its translations.